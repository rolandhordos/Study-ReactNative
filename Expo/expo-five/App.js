// @flow

import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'

// App is a top level component itself
export default class App extends Component {

  state = {
    count: 0,
  }

  componentDidMount() { // lifecycle - called once before initial render
    // We need a timer to advance the counter at a given rate
    setInterval(() => {          // bound function acting on this instance
      this.setState({ count: this.state.count + 1 })
    }, 1000)
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.counter}>{this.state.count}</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  counter: {
    fontSize: 60,
    color: 'blue',
  }
})
