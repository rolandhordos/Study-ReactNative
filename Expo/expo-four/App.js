import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'
import Carousel from '@rolandhordos/react-native-carousel'

export default class App extends Component {

  render() {
    return(
      <Carousel width={styles.slide.width} animate={false}>
        <View style={styles.slide}>
          <Text>Item 1</Text>
        </View>
        <View style={styles.slide}>
          <Text>Item 2</Text>
        </View>
        <View style={styles.slide}>
          <Text>Item 3</Text>
        </View>
      </Carousel>
    )
  }
}

const styles = StyleSheet.create({
  slide: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: 375,
    backgroundColor: '#00CED1', // Dark Turquoise
  }
})
