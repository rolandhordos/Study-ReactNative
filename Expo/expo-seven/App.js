// @flow

import React, { Component } from 'react'
import { View, Text } from 'react-native'

export default class Title extends Component<{}> {

  render() {
    return (
      <View>
        <Text>TODO List</Text>
      </View>
      )
    }

}
