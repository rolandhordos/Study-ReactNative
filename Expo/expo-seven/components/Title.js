// @flow

import React, { Component } from 'react'
import { View, Text } from 'react-native'
import styles from '../Style'

export default class Title extends Component<{}> {

  render() {
    return (
      <View style={styles.header}>
        <Text style={styles.title}>TODO List</Text>
      </View>
      )
    }

}
