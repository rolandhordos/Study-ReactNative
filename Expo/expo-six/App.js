// @flow

import React, { Component } from 'react'
import { View, Text, AppRegistry } from 'react-native'
import { createStore } from 'redux'

// Redux - Actions

const types = {
  INCREMENT: 'INCREMENT',
}

// Redux - Reducers

const reducer = (state, action) => {
  if (action.type === types.INCREMENT) {
    // return a complete representation of state
    return { count: state.count + 1 }
  }
  return state  // default course is a passthrough of input state
}

// Initialize the store
const initialState = {count: 0}
const store = createStore(reducer, initialState)

console.log('#Redux is ready')

// Throw some state changes at the store
store.dispatch({ type: types.INCREMENT })
store.dispatch({ type: types.INCREMENT })
store.dispatch({ type: types.INCREMENT })

export default class App extends Component<{}> {
  render() {
    return(
      <View style={{flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <Text style={{ fontSize: 100 }}>
        {store.getState().count}
      </Text>
      </View>
    )
  }
}
