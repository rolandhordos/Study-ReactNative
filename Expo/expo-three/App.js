import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default class App extends React.Component {
  render() {
    return (
      <View style={styling.container}><Text>Hello World</Text></View>
    );
  }
}

const styling = StyleSheet.create({
    container: {
      flexDirection: 'column',  // layout vertically from the top
      flex: 0.5,                // use only 50% of the screen
      justifyContent: 'center', // vertical alignment
      alignItems: 'center',     // horizontal alignment
    }
});
