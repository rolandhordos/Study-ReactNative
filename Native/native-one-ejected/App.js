import React from 'react';
import { View, Text, ScrollView, StyleSheet } from 'react-native';

export default class App extends React.Component {
  render() {
    return (
      <View style={styles.launchScreen}>
        <View style={styles.topHalfContainer}>
          <Text>Hello World</Text>
        </View>
        <ScrollView style={styles.bottomHalfContainer}>
          <View style={styles.box}/>
          <View style={styles.box}/>
          <View style={styles.box}/>
          <View style={styles.box}/>
          <View style={styles.box}/>
        </ScrollView>
      </View>
    );
  }
}

// Establish some normalized standards of User Experience.  Additionally if we use standard property names we can
// aggregate from DRY models using Object.assign.

// All boxes should use the same corner radius
const corner = { borderRadius: 8 }
// Establish a consistent container for half-screen layouts
const halfScreenContainer = {
  flexDirection: 'column',  // layout vertically from the top
  flex: 1.0,                // use only 50% of the screen, which for 2 sibling views means each are 100%
}
// The top half of the screen will hold the centered Hello World greeting
let greetingContainer = {
  justifyContent: 'center', // vertical alignment
  alignItems: 'center',     // horizontal alignment
  backgroundColor: 'whitesmoke',
}
Object.assign(greetingContainer, halfScreenContainer)  // adopt the standard half-screen settings

// Assembly time.
const styles = StyleSheet.create({
  launchScreen: {
    flex: 1.0,
  },
  topHalfContainer: greetingContainer,
  bottomHalfContainer: halfScreenContainer,
  box: {
    backgroundColor: 'orange',
    width: '30%',   // cool I can scale the layout for larger screens
    height: 110,
    margin: 12,
    borderRadius: corner.borderRadius,
  }
});
