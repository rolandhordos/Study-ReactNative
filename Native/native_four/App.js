import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'
import Carousel from '@rolandhordos/react-native-carousel'

export default class App extends Component {
  render() {
    return (
      <Carousel>
        <View style={styles.slide} width={styles.slide.width}>
          <Text>Slide 1</Text>
        </View>
        <View style={styles.slide} width={styles.slide.width}>
          <Text>Slide 2</Text>
        </View>
        <View style={styles.slide} width={styles.slide.width}>
          <Text>Slide 3</Text>
        </View>
      </Carousel>
    )
  }
}

const styles = StyleSheet.create({
  slide: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: 375,
  }
})
