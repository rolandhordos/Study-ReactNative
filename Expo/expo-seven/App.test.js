// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import App from './App'
import Title from './components/Title'

describe('the App', () => {

  const component = renderer.create(<App/>)
  const view = component.toJSON()
  const instance = component.getInstance()

  it('renders without crashing', () => {
    expect(component).toBeDefined()
    expect(view).toBeDefined()
    expect(instance instanceof App)
    // TODO: enable App snapshot matching when we're closer to assembling the entire app
    // expect(view).toMatchSnapshot()  // includes colors and alignment
  })

  it('has a Title', () => {
    const titleView = view.children[0]
    // const instance = component.toTree().instance
    expect(view.type).toBe('View')
    expect(titleView.type).toBe('Text')
  })

})
