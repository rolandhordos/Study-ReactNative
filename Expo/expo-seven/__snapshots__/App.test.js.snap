// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Title renders consistently 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    TODO List
  </Text>
</View>
`;
